internal/backend/db/dbgorm/sessions.go

```diff
@@ -128,11 +129,31 @@ func createLogRecord(db *gorm.DB, logr *scheme.SessionLogRecord) error {
 	// This is a hack in gorm to use subquery when creating an object.
 	// The goal is to have a sequence number per session.  The insert uses a subquery to get
 	// the current max sequence for this session_id and insert the next value in one query
+	// The loop is to handle optimistic locking, since concurrent writes can happen
+	// we have to retry on failure to commit
+	// We will try fo 10 times which should be enough retries
+	// This convention does not guarantee ordering of the log records
+	// Since inheritnly in the records them selfs, two records can happen in the same exact time
 	data["Seq"] = clause.Expr{SQL: "(select COALESCE(MAX(seq), 0)+1 from session_log_records where session_id = ?)", Vars: []interface{}{logr.SessionID}}
```

Comment:
I think the right solution is to use an `AUTOINCREMENT` field in the database.

Reply:
we can't use it in this case since we want to keep using sqlite/postgres and gorm
sqlite does not support auto increment column on a non primary key
and we actually want a sequence per session and not a generic sequence for all

we never going to query by sequence, only by (session_id,sequence) or just session_id

Reply:
OK then.